File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/date/CurrentDate.java
package coolpharaoh.tee.speicher.tea.timer.core.date;

import androidx.annotation.VisibleForTesting;

import java.util.Date;

public class CurrentDate {

    private CurrentDate() { }

    private static DateUtility dateUtility = new DateUtility();

    @VisibleForTesting
    public static void setFixedDate(final DateUtility fixedDate) {
        dateUtility = fixedDate;
    }

    public static Date getDate() {
        return dateUtility.getDate();
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/date/CurrentDate.kt
package coolpharaoh.tee.speicher.tea.timer.core.date

import androidx.annotation.VisibleForTesting
import java.util.Date

object CurrentDate {
    private var dateUtility = DateUtility()

    @JvmStatic
    @VisibleForTesting
    fun setFixedDate(fixedDate: DateUtility) {
        dateUtility = fixedDate
    }

    @JvmStatic
    fun getDate(): Date? {
        return dateUtility.getDate()
    }
}